### PR TITLE
Make spawn return Interpreter if called with a machine

### LINKED ISSUE #653: typescript useService(actor)
## Bug or feature request?
Bug?
## Description:
```js
export function Todo({ todoRef }) {
  const [state, send] = useService(todoRef);
  ...
```
On the TodoMVC example, we can call `useService` and pass an actor ( `todoRef`) , but with Typescript `Argument of type 'Actor<...>' is not assignable to parameter of type 'Interpreter<...>'.`

I searched in the comments and saw a [typescript implementation of todoMVC](https://codesandbox.io/s/pmx5j5n2p0), but it ditched the whole conflict and work around the problem, using `useMachine(Machine)` instead of `useService(Actor)`
## (Bug) Expected result:
I would like to use `useService(Actor)` in ts
## (Bug) Actual result:
 `Argument of type 'Actor<...>' is not assignable to parameter of type 'Interpreter<...>'.`
## (Bug) Potential fix:

Sorry I honestly do not know how to solve it, but if there's anything i could help let me know 👍

### PR DESCRIPTION
**Pull request by Andarist**
fixes #653

### CODE REVIEW

File: packages/xstate-react/test/types.test.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import { render } from '@testing-library/react';
+import { Machine, interpret, assign, spawn, Interpreter } from 'xstate';
+import { useService } from '../src';
+
+describe('useService', () => {
+  it('should accept spawned machine', () => {
```
Comment (Andarist) [inline packages/xstate-react/test/types.test.tsx]: it's somewhat verbose, but I've wanted to replicate the whole situation - from spawning to using service 🤷‍♂

File: test/actor.test.ts [hunk 2]
```diff
@@ -408,7 +408,7 @@ describe('actors', () => {
         initial: 'initial',
         states: {
           initial: {
-            entry: assign(() => ({
+            entry: assign((): { serverRef: Actor } => ({
```
Comment (Andarist) [inline test/actor.test.ts]: quite frankly I believe this shouldnt be needed but TS failed to "lose" information about the inferred type (Interpreter) when matching against expected one (Actor), even if Interpreter extends Actor

File: src/interpreter.ts [hunk 3]
```diff
@@ -1176,7 +1176,7 @@ const resolveSpawnOptions = (nameOrOptions?: string | SpawnOptions) => {
 export function spawn<TContext>(
   entity: StateMachine<TContext, any, any>,
   nameOrOptions?: string | SpawnOptions
-): Actor<State<TContext>>;
+): Interpreter<TContext>;
```
Comment (davidkpiano) [inline src/interpreter.ts]: This should remain `Actor` because `Interpreter` is not the only thing that can be spawned.

Comment (Andarist) [reply src/interpreter.ts]: This is just an overload - other spawnable types return Actor

Comment (davidkpiano) [reply src/interpreter.ts]: Ah, I see now 😅 Thanks!

**Pull request merged**